client/src/components/productDetails/eachComponent/Carousel.jsx

```diff
@@ -31,57 +54,195 @@ class Carousel extends React.Component {
     let second = this.state.current - 1;
     if (this.state.current === 0) {
       this.setState({
-        current: first,
+        current: this.state.current,
+      });
+    } else if (this.state.currentPage === 2 && this.state.listTwoCurrent !== 0) {
+      let newCurrent = this.state.current - 1;
+      let listTwoIndex = this.state.listTwoCurrent - 1;
+      document.getElementsByClassName('imageThumb')[this.state.listTwoCurrent].classList.remove('selectedThumbnail');
+      this.setState({
+        listTwoCurrent: listTwoIndex,
+        current: newCurrent,
+      }, () => {
+        document.getElementsByClassName('imageThumb')[this.state.listTwoCurrent].classList.add('selectedThumbnail');
+      });
+    } else if (this.state.currentPage === 2 && this.state.listTwoCurrent === 0) {
+      let newCurrent = this.state.current - 1;
```

Comment:
Try to use Const instead of Let for variables that will not be directly altered elsewhere